### PR TITLE
Enable `clippy::undocumented_unsafe_blocks` warning across the workspace

### PR DESCRIPTION
**Pull request by 13ros27**
# Objective

Enables warning on `clippy::undocumented_unsafe_blocks` across the workspace rather than only in `bevy_ecs`, `bevy_transform` and `bevy_utils`. This adds a little awkwardness in a few areas of code that have trivial safety or explain safety for multiple unsafe blocks with one comment however automatically prevents these comments from being missed.

## Solution

This adds `undocumented_unsafe_blocks = "warn"` to the workspace `Cargo.toml` and fixes / adds a few missed safety comments. I also added `#[allow(clippy::undocumented_unsafe_blocks)]` where the safety is explained somewhere above.

There are a couple of safety comments I added I'm not 100% sure about in `bevy_animation` and `bevy_render/src/view` and I'm not sure about the use of `#[allow(clippy::undocumented_unsafe_blocks)]` compared to adding comments like `// SAFETY: See above`.

### CODE REVIEW

File: crates/bevy_animation/src/lib.rs [hunk 1]
```diff
@@ -646,6 +646,7 @@ fn apply_animation(
             let Ok(mut transform) = (unsafe { transforms.get_unchecked(target) }) else {
                 continue;
             };
+            #[allow(clippy::undocumented_unsafe_blocks)]
```
Comment (alice-i-cecile) [inline crates/bevy_animation/src/lib.rs:649]: I'd like to fix this one in this PR.

File: crates/bevy_window/src/raw_handle.rs [hunk 2]
```diff
@@ -34,6 +34,7 @@ impl RawHandleWrapper {
 // A recommendation for this pattern (and more context) is available here:
 // https://github.com/rust-windowing/raw-window-handle/issues/59
 unsafe impl Send for RawHandleWrapper {}
+#[allow(clippy::undocumented_unsafe_blocks)]
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/raw_handle.rs:37]: This should probably just reference the safety comment for `Send`

File: crates/bevy_render/src/render_resource/resource_macros.rs [hunk 3]
```diff
@@ -58,6 +58,7 @@ macro_rules! render_resource_wrapper {
         // If in future there is a case where a wrapper is required for a non-send/sync type
         // we can implement a macro variant that omits these manual Send + Sync impls
         unsafe impl Send for $wrapper_type {}
+        #[allow(clippy::undocumented_unsafe_blocks)]
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/render_resource/resource_macros.rs:61]: I would probably just duplicate this safety comment.

File: crates/bevy_ui/src/ui_node.rs [hunk 4]
```diff
@@ -1309,6 +1309,7 @@ pub struct GridPlacement {
 impl GridPlacement {
     pub const DEFAULT: Self = Self {
         start: None,
+        #[allow(clippy::undocumented_unsafe_blocks)]
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/ui_node.rs:1312]: This should be fixed in this PR.

File: crates/bevy_tasks/src/task_pool.rs [hunk 5]
```diff
@@ -353,12 +353,16 @@ impl TaskPool {
         // Any usages of the references passed into `Scope` must be accessed through
         // the transmuted reference for the rest of this function.
         let executor: &async_executor::Executor = &self.executor;
+        #[allow(clippy::undocumented_unsafe_blocks)]
```
Comment (alice-i-cecile) [inline crates/bevy_tasks/src/task_pool.rs:356]: Ideally we can fix all of these in this PR.

File: crates/bevy_animation/src/lib.rs (hunk 1 above)
Comment (13ros27) [reply crates/bevy_animation/src/lib.rs]: I've replaced the various `#[allow(...)]` with safety comments referencing the safety comments above them and reiterating in brief

**Review by alice-i-cecile — APPROVED** [review]

**Review by killercup — APPROVED** [review]

**Pull request merged**